src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -154,6 +159,64 @@ internal partial class Binder
             { WasCompilerGenerated = wasCompilerGenerated };
         }
 
+        private BoundExpression ConvertObjectCreationExpression(BoundUnconvertedObjectCreationExpression node, bool isCast, TypeSymbol destination, DiagnosticBag diagnostics)
+        {
+            var arguments = AnalyzedArguments.GetInstance(node.Arguments, node.ArgumentRefKindsOpt, node.ArgumentNamesOpt);
+            BoundExpression expr = BindObjectCreationExpression(node, destination.StrippedType(), arguments, diagnostics);
+            if (destination.IsNullableType())
+            {
+                // We manually create an ImplicitNullable conversion
+                // if the destination is nullable, in which case we
+                // target the underlying type e.g. `S? x = new();`
+                // is actually identical to `S? x = new S();`.
+                var conversion = new Conversion(ConversionKind.ImplicitNullable, Conversion.IdentityUnderlying);
```

Comment:
We should instead use `Conversion.ClassifyStandardConversion` to create this, to ensure that we pick up any use-site diagnostics.